# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/blink/FFTConvolver.cpp
# Commit: e519654e5d8b
# Full Hash: e519654e5d8b7e0dc682c316fa0d16318b962765
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-28 04:30:28
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Scale the inverse FFT result using tx instead of pre- or post-processing the data. r=karlt
#   
#   This skips a copy at the expense of having to do a `const_cast`.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202434
# ==============================================================================

diff -r 9b17ac4800c6 -r e519654e5d8b dom/media/webaudio/blink/FFTConvolver.cpp
--- a/dom/media/webaudio/blink/FFTConvolver.cpp	Tue Feb 27 12:40:23 2024 +0000
+++ b/dom/media/webaudio/blink/FFTConvolver.cpp	Tue Feb 27 12:40:24 2024 +0000
@@ -85,7 +85,7 @@
     // The input buffer is now filled (get frequency-domain version)
     m_frame.PerformFFT(m_inputBuffer.Elements());
     m_frame.Multiply(*fftKernel);
-    m_frame.GetInverseWithoutScaling(m_outputBuffer.Elements());
+    m_frame.GetInverse(m_outputBuffer.Elements());
 
     // Overlap-add 1st half from previous time
     AudioBufferAddWithScale(m_lastOverlapBuffer.Elements(), 1.0f,